cartoframes/dataset_info.py

```diff
@@ -0,0 +1,66 @@
+import time
+
+from carto.datasets import DatasetManager
+from carto.exceptions import CartoException
+
+
+class DatasetInfo():
+    PRIVATE = 'PRIVATE'
+    PUBLIC = 'PUBLIC'
+    LINK = 'LINK'
+
+    def __init__(self, carto_context, table_name):
+        self._get_metadata(carto_context, table_name)
```

Comment:
I think is better to explicitly assign a variable and not assign state to instance variables inside a method. This could lead to side-effects in the near-future (is just an opinion)

Reply:
I agree